### DISCUSSION TITLE
Warning while using SwitchStore

### DISCUSSION THREAD

**Discussion by Ramesh1402** [pointfreeco/swift-composable-architecture#2366] [General]
### Description

Issue : While using SwitchStore I am getting a warning saying "Mismatch between cases being switched on". please refer the screenshot attached.

Framework Version: 1.0.0
OS Version: iOS 16.4

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

There should be no warning when using SwitchStore with appropriate cases.

### Actual behavior

Warning being displayed when using SwitchStore. 

### Steps to reproduce

1. Create a plain swiftUi Project 
2. add swift composable architecture 1.0.0  as a dependency.
3. Use SwitchStore to switch between different views in a container view.

Code_ScreenShot:
<img width="1133" alt="Sample_Code" src="https://github.com/pointfreeco/swift-composable-architecture/assets/24583016/24cfcc67-f92d-4081-a610-f6b21a1503ad">

Error ScreenShot:
![Error_ScreenShot](https://github.com/pointfreeco/swift-composable-architecture/assets/24583016/4d3d71a7-d9f5-480c-a0ff-6d084d3908c1)

### The Composable Architecture version information

1.0.0

### Destination operating system

iOS 16.4

### Xcode version information

Xcode 14.3.1

### Swift Compiler version information

```shell
swift-driver version: 1.75.2 Apple Swift version 5.8.1 (swiftlang-5.8.0.124.5 clang-1403.0.22.11.100)
Target: arm64-apple-macosx13.0
```

**Comment by mbrandonw** [comment]
Hi @Ramesh1402, your use of `SwitchStore` is slightly wrong. If you look at the [docs](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/switchstore) you will see it should be used like this:

```swift
SwitchStore(store: …) { 
  switch $0 {
    case .login: 
      CaseLet(…)
    case .home: 
      CaseLet(…)
  }
}
```

Since I don't think this is an issue with the library I am going to convert it to a discussion. Feel free to follow up with more questions there.

**Reply by Ramesh1402** [reply]
Thank you for the answer!
